jupyter-org-extensions.el

```diff
@@ -0,0 +1,401 @@
+    (end-of-line)))
+
+;;;###autoload
+(defun jupyter-org-merge-blocks ()
+  "Merge the current block with the next block."
+  (interactive)
+  (unless (org-in-src-block-p)
+    (error "Not in a source block"))
+  (let ((current-src-block (org-element-context)))
+    (org-babel-remove-result)
+    (org-babel-next-src-block)
+    (let* ((next-src-block (prog1 (org-element-context)
+                   (org-babel-remove-result)))
+           (next-src-block-beg (set-marker
+                      (make-marker)
+                      (org-element-property :begin next-src-block)))
+           (next-src-block-end (set-marker
+                      (make-marker)
+                      (jupyter-org-element-end-before-blanks next-src-block))))
+      (goto-char (jupyter-org-element-end-before-blanks current-src-block))
+      (forward-line -1)
+      (insert
+       (delete-and-extract-region
+        (save-excursion
+          (goto-char (jupyter-org-element-begin-after-affiliated next-src-block))
+          (forward-line 1)
+          (point))
+        (save-excursion
+          (goto-char next-src-block-end)
+          (forward-line -1)
+          (point))))
+      ;; delete a leftover space
+      (save-excursion
+        (goto-char next-src-block-end)
+        (when (looking-at-p "[[:space:]]*$")
+          (set-marker next-src-block-end (+ (point-at-eol) 1))))
+      (delete-region next-src-block-beg next-src-block-end)
+      (set-marker next-src-block-beg nil)
+      (set-marker next-src-block-end nil)))
+  ;; move to the end of the last line
+  (forward-line -1)
+  (end-of-line))
+
+;;;###autoload
+(defun jupyter-org-move-src-block (&optional below)
+  "Move source block before of after another.
+
+If BELOW is non-nil, move the block down, otherwise move it up."
+  (interactive)
+  (unless (org-in-src-block-p)
+    (error "Not in a source block"))
+  ;; throw error if there's no previous or next source block
+  (when (ignore-errors
+          (save-excursion
+            (if below
+                (org-babel-next-src-block)
+              (org-babel-previous-src-block))))
+    (let* ((region (jupyter-org-select-block-and-results))
+           (block (delete-and-extract-region (car region) (cdr region))))
+      ;; if there is an empty line remaining, take that line as part of the
+      ;; ... block
+      (when (and (looking-at-p "[[:space:]]*$") (/= (point) (point-max)))
+        (delete-region (point-at-bol) (+ (point-at-eol) 1))
+        (setq block (concat block "\n")))
+      (if below
+          ;; if below, move past the next source block or its result
+          (let ((next-src-block-head (org-babel-where-is-src-block-head)))
+            (if next-src-block-head
+                (goto-char next-src-block-head)
+              (org-babel-next-src-block))
+            (let ((next-src-block (org-element-context))
+                  (next-results-start (org-babel-where-is-src-block-result)))
+              (if next-results-start
```

Comment:
What about `(if (null ...` to get rid of the `progn`?

Reply:
Sure, can be done.